lib/api/controllers/indexController.js

```diff
@@ -1,100 +1,121 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2017 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 'use strict';
 
-var
+const
   _ = require('lodash'),
-  Promise = require('bluebird'),
+  Bluebird = require('bluebird'),
   Request = require('kuzzle-common-objects').Request,
-  assertHasBody = require('./util/requestAssertions').assertHasBody,
-  assertBodyHasAttribute = require('./util/requestAssertions').assertBodyHasAttribute,
-  assertBodyAttributeType = require('./util/requestAssertions').assertBodyAttributeType,
-  assertHasIndex = require('./util/requestAssertions').assertHasIndex;
+  {
+    assertHasBody,
+    assertBodyHasAttribute,
+    assertBodyAttributeType,
+    assertHasIndex
+  } = require('./util/requestAssertions');
 
 /**
+ * @class IndexController
  * @param {Kuzzle} kuzzle
- * @constructor
  */
-function IndexController (kuzzle) {
-  /** @type ElasticSearch */
-  var engine = kuzzle.services.list.storageEngine;
+class IndexController {
+  constructor(kuzzle) {
+    /** @type Kuzzle */
+    this.kuzzle = kuzzle;
+
+    /** @type ElasticSearch */
+    this.engine = kuzzle.services.list.storageEngine;
+  }
 
   /**
    * Reset all indexes
    *
    * @param {Request} request
    * @returns {Promise<Object>}
    */
-  this.mDelete = function indexMDelete (request) {
-    var
-      allowedIndexes = [],
-      indexes;
+  mDelete(request) {
+    const allowedIndexes = [];
 
     assertHasBody(request);
     assertBodyHasAttribute(request, 'indexes');
 
-    return engine.listIndexes()
+    return this.engine.listIndexes()
       .then(res => {
-        var promises;
+        const indexes = res.indexes.filter(index => _.includes(request.input.body.indexes, index));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 99 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)